### ISSUE TITLE
[pfcwd] pfcwd does not consider the change of pfc configuraion on port

### ISSUE THREAD

**Issue by leoli-nps** [Azure/sonic-swss#842]
**Description**

When testing the case described in section `2.6.3 Queue Reconfiguration or Removal` in the document `PFC-Watchdog-Design`, we found that whenever enable PFCWD on port, pfcwdorch will init PFCWD counters only on which queues enabled PFC; After that , PFCWD no longer watch the PFC configuration on port; If a user disable PFC on a queue or enable PFC on a new queue, PFCWD would not know about that and update accordingly. Finally, although PFC is disabled on the queue that was marked as stormed, but the queue failed to be restored.

In addition, during the test, we can't directly enable or disable PFC on a queue, but indirectly enable and disable PFC by the way shown below.

We put the PFC configuration in the file `/etc/sonic/pfc.json`, and then enable and disable the pfc on the port by modifying the file. Finally, use the command `sudo config load /etc/sonic/pfc.json` to make the configuration take effect.

For example, initially, pfc is enabled on queue 0/1 of Ethernet 120, as shown below：
```
admin@sonic:~$ sudo vim /etc/sonic/pfc.json
{
    "PORT_QOS_MAP": {
        "Ethernet120": {
            "pfc_enable": "0,1",
            "pfc_to_queue_map": "[MAP_PFC_PRIORITY_TO_QUEUE|AZURE]",
            "pfc_to_pg_map": "[PFC_PRIORITY_TO_PRIORITY_GROUP_MAP|AZURE]"
        }
    }
}
```
Now need to disable them, and the file /etc/sonic/pfc.json is modified as follows：
```
admin@sonic:~$ sudo vim /etc/sonic/pfc.json
{
    "PORT_QOS_MAP": {
        "Ethernet120": {
            "pfc_enable": "2",
            "pfc_to_queue_map": "[MAP_PFC_PRIORITY_TO_QUEUE|AZURE]",
            "pfc_to_pg_map": "[PFC_PRIORITY_TO_PRIORITY_GROUP_MAP|AZURE]"
        }
    }
}
```

In summary, I think the following two issues should be resolved:
1. A command to enable/disable PFC on a port should be provided.
2. A mechanism should be introduced to let PFCWD respond to this event.

Signed-off-by: leo.li leo.li@nephosinc.com

**Comment by wendani** [comment]
Do you see an application need to enable/disable PFC on a queue/pg?

**Comment by leoli-nps** [comment]
@wendani, as you said, at the moment I really don't see a specific application that needs to enable/disable PFC on a queue/pg. However, the issue is raised, because the following description is seen in the document [PFC-Watchdog-Design](https://github.com/Azure/SONiC/wiki/PFC-Watchdog-Design#263-queue-reconfiguration-or-removal).
```
2.6.3 Queue Reconfiguration or Removal
Queue configuration must be going through PFC WD proxy in order to make it update its internal state. If PFC is disabled on a queue that was marked as stormed, queue will be restored.
```
We tested according to this description, but the test results did not meet the expectations in the description.
